subsys/bluetooth/host/gatt.c

```diff
@@ -2130,23 +2138,37 @@ static int gatt_notify_mult(struct bt_conn *conn, uint16_t handle,
 	}
 
 	if (!*buf) {
-		*buf = bt_att_create_pdu(conn, BT_ATT_OP_NOTIFY_MULT,
-					 sizeof(*nfy) + params->len);
+		*buf = bt_att_create_pdu(conn, BT_ATT_OP_NOTIFY_MULT, pdu_len);
 		if (!*buf) {
 			return -ENOMEM;
 		}
 
-		bt_att_set_tx_meta_data(*buf, params->func, params->user_data);
+		bt_att_set_tx_meta_data(*buf, tuple[0].func, tuple[0].user_data);
 	}
 
-	BT_DBG("handle 0x%04x len %u", handle, params->len);
+	for (i = 0; i < num_params; i++) {
+		/* Check if attribute is a characteristic then adjust the handle */
+		if (!bt_uuid_cmp(tuple[i].attr->uuid, BT_UUID_GATT_CHRC)) {
+			struct bt_gatt_chrc *chrc = tuple[i].attr->user_data;
 
-	nfy = net_buf_add(*buf, sizeof(*nfy));
-	nfy->handle = sys_cpu_to_le16(handle);
-	nfy->len = sys_cpu_to_le16(params->len);
+			if (!(chrc->properties & BT_GATT_CHRC_NOTIFY)) {
+				return -EINVAL;
+			}
 
-	net_buf_add(*buf, params->len);
-	memcpy(nfy->value, params->data, params->len);
+			handle = bt_gatt_attr_value_handle(tuple[i].attr);
+		} else {
+			handle = bt_gatt_attr_get_handle(tuple[i].attr);
+		}
+
+		BT_DBG("handle 0x%04x len %u", handle, tuple[i].len);
+
+		nfy = net_buf_add(*buf, sizeof(*nfy));
+		nfy->handle = sys_cpu_to_le16(handle);
+		nfy->len = sys_cpu_to_le16(tuple[i].len);
+
+		net_buf_add(*buf, tuple[i].len);
+		memcpy(nfy->value, tuple[i].data, tuple[i].len);
```

Comment:
`ATT_MULTIPLE_HANDLE_VALUE_NTF` is defined as having a minimum size of 1 + 8, but since the format is (2 + 2 + 0 or above), does that implicitly mean that we shall notify at least 2 handles to use this? Or what does the minimum 8 octets come from? 

![image](https://user-images.githubusercontent.com/2520413/168581886-93ff7e18-a0c6-4555-9d5c-bba350aead87.png)

Reply:
>  does that implicitly mean that we shall notify at least 2 handles to use this?

Yes, I think it is not allowed to use this PDu with a single handle.

> A server can send a notification of two or more attributes' values at any time.

Reply:
That makes sense. Then we also have a bug in the current implementation :D (as well as in this PR at this moment).